addons/account/models/account_move.py

```diff
@@ -2227,6 +2227,8 @@ def post(self):
         if not self.env.su and not self.env.user.has_group('account.group_account_invoice'):
             raise AccessError(_("You don't have the access rights to post an invoice."))
         for move in self:
+            if move.state == 'posted':
+                raise UserError(_('This move is already posted.'))
```

Comment:
Rather 
This entry (%(move_name)s (id %(move_id)s) is already posted. 

Adding information about the faulty object is always good in this case ; it could happen in a more complex flow, with records being treated by batch. This way, we can find the culprit more easily :)

Reply:
Ok, change applied.